src/Microsoft.Azure.WebJobs.Host.Storage/Singleton/BlobLeaseDistributedLockManager.cs

```diff
@@ -93,61 +89,59 @@ public async virtual Task<string> GetLockOwnerAsync(string account, string lockI
                 await WriteLeaseBlobMetadata(lockBlob, leaseId, lockOwnerId, cancellationToken);
             }
 
-            SingletonLockHandle lockHandle = new SingletonLockHandle(leaseId, lockId, lockBlob, lockPeriod);
+            SingletonLockHandle lockHandle = new SingletonLockHandle(leaseId, lockId, this.GetBlobLeaseClient(lockBlob, leaseId), lockPeriod);
 
             return lockHandle;
         }
 
-        protected abstract CloudBlobContainer GetContainer(string accountName);
-
-        internal CloudBlobDirectory GetLockDirectory(string accountName)
+        protected virtual BlobContainerClient GetContainerClient(string accountName)
         {
-            if (string.IsNullOrEmpty(accountName))
+            if (!string.IsNullOrWhiteSpace(accountName))
             {
-                accountName = string.Empty; // must be non-null for a dictionary key
+                throw new InvalidOperationException("Must replace singleton lease manager to support multiple accounts");
             }
 
-            CloudBlobDirectory storageDirectory = null;
-            if (!_lockDirectoryMap.TryGetValue(accountName, out storageDirectory))
-            {
-                var container = this.GetContainer(accountName);
+            return _azureStorageProvider.GetWebJobsBlobContainerClient();
+        }
 
-                storageDirectory = container.GetDirectoryReference(HostDirectoryNames.SingletonLocks);
-                _lockDirectoryMap[accountName] = storageDirectory;
-            }
+        internal string GetLockPath(string lockId)
+        {
+            // lockId here is already in the format {accountName}/{functionDescriptor}.{scopeId}
```

Comment:
This comment isn't entirely correct, I'd just remove it. E.g. lockId is formatted by SingletonManager.FormatLockId and is {hostId}/{functionName}.{scopeId} for example. No need to repeat that here I think.